### PR TITLE
fix: apply enum for task states and fix candidate group and user

### PR DESCRIPTION
**Pull request by nathansandi**
## Description

- Rename `candidateGroup` to `candidateGroups` 
- Rename `candidateUser` to `candidateUsers`
- Create `enums` for User Task State 
- Update Client and Controller tests

### CODE REVIEW

File: zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryResponseMapper.java [hunk 1]
```diff
@@ -345,10 +345,10 @@ public static UserTaskItem toUserTask(final UserTaskEntity t) {
         .processDefinitionKey(t.processDefinitionId())
         .elementInstanceKey(t.flowNodeInstanceId())
         .processDefinitionId(t.bpmnProcessId())
-        .state(t.state())
+        .state(UserTaskItem.StateEnum.fromValue(String.valueOf(t.state())))
```
Comment (tmetzke) [inline zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryResponseMapper.java:348]: 🔧 I haven't looked into this in my IDE but it feels like this could be done in a simpler way. What does `.state()` expect as a value and what is the type of `t.state()`?

Comment (nathansandi) [reply zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryResponseMapper.java]: Yes it possible to make it simpler, this expects receive a UserItem.StateEnum

I have updated the code on this way now: `.state(UserTaskItem.StateEnum.fromValue(t.state().name()))`

Comment (tmetzke) [reply zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryResponseMapper.java]: The type of `t.state()` is a different enum class, I assume?

Comment (nathansandi) [reply zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryResponseMapper.java]: Yes ;) It is from `UserItem`, now its like in the same way as the others.

Comment (tmetzke) [reply zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryResponseMapper.java]: OK, I figured that, yeah. Good enough for me 👍

File: search/search-domain/src/main/java/io/camunda/search/entities/UserTaskEntity.java [hunk 2]
```diff
@@ -19,7 +19,7 @@ public final record UserTaskEntity(
     String creationTime,
     String completionTime,
     String assignee,
-    String state,
+    UserTaskState state,
```
Comment (tmetzke) [inline search/search-domain/src/main/java/io/camunda/search/entities/UserTaskEntity.java:22]: ❓ Is this change really necessary? In the search domain, it could also just stay a String, right? This should only be about the REST API, where more "type safety" would be nice for users.

Comment (nathansandi) [reply search/search-domain/src/main/java/io/camunda/search/entities/UserTaskEntity.java:22]: It works without this change as well, I follow up what was develop in another entities as:
- ProcessInstanceEntity
- FlowNodeInstanceEntity
I have applied on UserTaskEntity to keep the consistency

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 3]
```diff
@@ -2583,18 +2588,23 @@ components:
           format: int64
         state:
           type: string
```
Comment (tmetzke) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:2590]: 🔧 By the way, all of the user task search components don't have descriptions. That makes it harder for users to understand the properties.

File: search/search-domain/src/main/java/io/camunda/search/entities/UserTaskEntity.java (hunk 2 above)
Comment (tmetzke) [reply search/search-domain/src/main/java/io/camunda/search/entities/UserTaskEntity.java:22]: Interesting, I wasn't aware 👍 
I'll have a look at that as well. This is non-blocking 👍

**Review by tmetzke — APPROVED** [review]
👍 The changes are valid, thanks, @nathansandi.
🔧 There are some smaller things we can follow-up on if we don't want to tackle them right away, like adding more descriptions in the OpenAPI spec.

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml (hunk 3 above)
Comment (nathansandi) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml:2603]: Sounds great :) 
I have added the more descriptions here https://github.com/camunda/camunda/pull/23598#:~:text=on%20rest%2Dapi-,33b21de,-Merge%20state 

I am going to tag a tech written for review ;).

**Review by christinaausley — APPROVED** [review]
Uppercased ID, otherwise looks good!
